docs/guides/transfer_learning_full.md

```diff
@@ -0,0 +1,370 @@
+  * Creating a model for the new task
+  * Transfering pretrained parameters
+* Optimization Procedure
+  * Discuss Options
+    * Using `stop_gradient`
+    * Taking gradients w.r.t. trainable parameters only
+    * Zeroing out gradients for non-trainable parameters
+  * Using Optax `multi_transform` + `variables_path_partitions`
+    * Completely freezing layers
+    * Using different learning rates
+* Training the model
+
++++
+
+## Introduction
+
+### What is Transfer Learning?
+Theory
+* Pretrain on a base task, use the learned parameters to train on a target task
+* Works if the representation learned on the base task are useful for the target task
+* Usually done when you have a small dataset for the target task
+* The more general base task is the higher the chance that the representations learned are useful for the target task, models trained via Self Supervised Learning (e.g `BERT`) are good candidates.
+
+Options in Practice
+* Fine-tuning: use the pretrained weights as a starting point but train the whole model on the new task. This is commonly done in NLP tasks.
+* Extract features: use the pretrained model to extract the features for the target task and train a new model on those features. This is commonly done in vision tasks. (mention pre-computing all embeddings as an optimization)
+* Hybrid: you can interpolate between the two previous approaches by deciding how much of the pretrained model remains frozen and how much is trained, e.g. freeze the last few layers and train the rest. Apart from that you can also interpolate between freezing and training by using different learning rates per layer where freezing is just the extreme case of the learning rate being 0.
+
+### Project Description
+For this tutorial we will be creating a model for a simple Computer Vision classification task on a small dataset, more specifically we will using a pretrained `MlpMixer` Flax model from the [vision_transformer](https://github.com/google-research/vision_transformer) repository to learn the [rock_paper_scissors](https://www.tensorflow.org/datasets/catalog/rock_paper_scissors) dataset from TensorFlow Datasets.
+
+## Creating the model
+The first thing you need to do is load a pretrained, depending on which model/library you are using this might look wildly different but in general you will have to get hold of the following:
+
+* The Flax `Module` that defines the model architecture.
+* The model parameters (and batch statistics for models that use batch norm).
+* The preprocessing procedure that converts the raw inputs into the format the model expects, for images this is often a simple processing scripts that applies resize and some normalization, and for text this usually a tokenizer.
+
+In this tutorial we will create an `MlpMixer` model using the `vit_jax` library from the `vision_transformer` repository. The thing we need to do is to install the `vit_jax` library:
+
+```{code-cell} ipython3
+! pip install transformers[flax]
```

Comment:
```suggestion
!pip install transformers[flax]
```

Reply:
Also `!pip install -U flax jax jaxlib` since `transformers` uses different versions of Flax and JAX.